cni-plugin/pkg/dataplane/linux/dataplane_linux.go

```diff
@@ -82,16 +89,17 @@ func (d *linuxDataplane) DoNetworking(
 		la.NumTxQueues = d.queues
 		la.NumRxQueues = d.queues
 		veth := &netlink.Veth{
-			LinkAttrs: la,
-			PeerName:  hostVethName,
+			LinkAttrs:     la,
+			PeerName:      hostVethName,
+			PeerNamespace: netlink.NsFd(int(hostNS.Fd())),
 		}
 
 		if err := netlink.LinkAdd(veth); err != nil {
 			d.logger.Errorf("Error adding veth %+v: %s", veth, err)
 			return err
 		}
 
-		hostVeth, err := netlink.LinkByName(hostVethName)
+		hostVeth, err := hostNlHandle.LinkByName(hostVethName)
```

Comment:
I'm wondering how this is working given that `hostVethName` is already in another namespace.

Reply:
The netlink socket created by netlink.NewHandle() and used by hostNlHandle is tied to the namespace where it was created.

Reply:
Got it now, thanks.